### PR TITLE
Beautify the examples

### LINKED ISSUE #676: Mixing plotting + animations + simulations produces bad examples
Since PR #621 introduced simulations, the examples have actually started looking worse as plotting + animation + simulations are all mixed together.

Simplest example from https://climate-machine.github.io/Oceananigans.jl/stable/generated/one_dimensional_diffusion/
```julia
anim = @animate for i=1:100
    simulation.stop_iteration += 100
    run!(simulation)

    plot(interior(T)[1, 1, :], zC, linewidth=2, title=tracer_label(model),
         label="", xlabel="Tracer concentration", ylabel="z", xlims=(0, 1))
end

mp4(anim, "1d_diffusion.mp4", fps = 15) # hide
```

I don't think that's a clean use of the simulation class. I think we should add an output writer, run the simulation, then have a separate loop that reads the output and plots/animates it.

### PR DESCRIPTION
**Pull request by glwagner**
Some of the examples had typos and incorrect comments. Also, some of the plotting is improved now that we know beautiful plots means `contourf` with `linewidth=0`. I added a few features to some of the examples, most notably `AveragedField`. Finally, we now always visualize in post-process.

Depends on #1093, since that PR beautifies a few of the examples as well with field-dependent boundary conditions.

Resolves #676

### CODE REVIEW

**Comment by navidcy** [comment]
@glwagner seems like travis has trouble building the docs... did you try building locally and it was OK?

**Comment by glwagner** [comment]
> I guess this is due to #1093?

Good catch; I've gotten rid of it.

**Comment by navidcy** [comment]
Still errors... travis complains that Doc build takes too long. :(

When docs built they should be available for preview at:
https://clima.github.io/OceananigansDocumentation/previews/PR1099

**Comment by glwagner** [comment]
I should probably shorten some of the simulations and movies then...

**Comment by glwagner** [comment]
I guess `contourf` takes a lot longer than `heatmap`.

**Comment by navidcy** [comment]
The examples looks great!

File: examples/convecting_plankton.jl [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+# ## Simulation setup
+#
+# We use a `TimeStepWizard` that limits the
+# time-step to 1 minute, and adapts the time-step such that CFL
+# (Courant-Freidrichs-Lewy) number remains below 1.0,
+
+using Oceananigans.Utils: minute
+
+wizard = TimeStepWizard(cfl=1.0, Δt=2minute, max_change=1.1, max_Δt=2minute)
```
Comment (ali-ramadhan) [inline examples/convecting_plankton.jl:91]:
```suggestion
wizard = TimeStepWizard(cfl=1.0, Δt=2minutes, max_change=1.1, max_Δt=2minutes)
```

File: examples/ocean_wind_mixing_and_convection.jl [hunk 2]
```diff
@@ -109,75 +132,133 @@ nothing # hide
...
-    w_plot = heatmap(xC, zF, w', xlabel="x (m)", ylabel="z (m)", color=:balance, clims=(-3e-2, 3e-2))
-    T_plot = heatmap(xC, zC, T', xlabel="x (m)", ylabel="z (m)", color=:thermal, clims=(19.75, 20.0))
-    S_plot = heatmap(xC, zC, S', xlabel="x (m)", ylabel="z (m)", color=:haline,  clims=(34.99, 35.01))
+    kwargs = (linewidth=0, xlabel="x (m)", ylabel="z (m)", aspectratio=1,
+              xlims=(0, grid.Lx), ylims=(-grid.Lz, 0))
 
+    w_plot = contourf(xw, zw, w'; color=:balance, clims=wlims, levels=wlevels, kwargs...)
+    T_plot = contourf(xT, zT, T'; color=:thermal, clims=Tlims, levels=Tlevels, kwargs...)
+    S_plot = contourf(xT, zT, S'; color=:haline,  clims=Slims, levels=Slevels, kwargs...)
+    ν_plot = heatmap(xT, zT, νₑ'; color=:thermal, clims=νlims, levels=νlevels, kwargs...)
```
Comment (ali-ramadhan) [inline examples/ocean_wind_mixing_and_convection.jl:250]:
```suggestion
    ν_plot = contourf(xT, zT, νₑ'; color=:thermal, clims=νlims, levels=νlevels, kwargs...)
```

Would this look better with `contourf`?

image

Comment (ali-ramadhan) [reply examples/ocean_wind_mixing_and_convection.jl:250]: Ah just saw your comment about `contourf` being slower :(

**Review by ali-ramadhan — APPROVED** [review]
Examples are beautiful indeed!

File: examples/ocean_wind_mixing_and_convection.jl (hunk 2 above)
Comment (glwagner) [reply examples/ocean_wind_mixing_and_convection.jl:250]: Ah, there's another reason to use heatmap here --- I think it's actually useful to see the grid point values of the eddy viscosity. I'll add a comment about this in the example.

**Comment by navidcy** [comment]
Any ideas for #944? Would be nice to deal with it here if possible..

**Comment by glwagner** [comment]
Switching to gif seems like the only solution.

**Comment by navidcy** [comment]
> Switching to gif seems like the only solution.

gifs are larger in size but if it can be handled by Travis or whatever builds the docs then why not?

Btw, all this docs building and previewing (for this PR and all the rest) might be getting the http://github.com/CliMA/OceananigansDocumentation repo size out of hand. Is there a limit by Github? If not, then we should not worry... But if there is, then one could clean up the [gh-pages/previews](https://github.com/CliMA/OceananigansDocumentation/tree/gh-pages/previews) folder with [bfg-repo-cleaner](https://rtyley.github.io/bfg-repo-cleaner/) at some point.

**Comment by ali-ramadhan** [comment]
Sounds like we should be fine up until maybe 5 GB after which GitHub might get angry with us haha: https://docs.github.com/en/free-pro-team@latest/github/managing-large-files/what-is-my-disk-quota#file-and-repository-size-limitations

**Comment by navidcy** [comment]
Nice. Then when/if GitHub indeed tries to twist your arm @ali-ramadhan you can respond asking them to delete all folders in the `previews` directory + their history. Then nobody needs to bother with `bfg-cleaner` which might be cumbersome...

**Pull request merged**